src/routes/safe/components/Apps/components/AddAppForm/AppAgreement.tsx

```diff
@@ -9,27 +9,45 @@ import Field from 'src/components/forms/Field'
 const StyledCheckbox = styled(Checkbox)`
   margin: 0;
 `
+const StyledAgreementContainer = styled.div`
+  display: flex;
+  flex: 1;
+  .MuiFormControlLabel-root {
+    margin-right: 0;
+  }
+`
 
 const AppAgreement = (): React.ReactElement => {
   const { visited } = useFormState({ subscription: { visited: true } })
-
+  const [checked, setCheck] = useState(false)
   // trick to prevent having the field validated by default. Not sure why this happens in this form
   const validate = !visited?.agreementAccepted ? undefined : required
 
+  const handleOnChange = ({ onBlur, onFocus, onChange }, event, value: boolean) => {
+    onFocus(event)
+    onChange(event)
+    onBlur(event)
+    setCheck(value)
+  }
+
   return (
-    <Field
-      component={StyledCheckbox}
-      label={
-        <Text size="xl">
-          This app is not a Gnosis product and I agree to use this app
-          <br />
-          at my own risk.
-        </Text>
-      }
-      name="agreementAccepted"
-      type="checkbox"
-      validate={validate}
-    />
+    <Field name="agreementAccepted" type="checkbox" validate={validate}>
+      {({ input }) => (
+        <StyledAgreementContainer>
+          <StyledCheckbox
+            checked={checked}
+            name="agreementAccepted"
+            label=""
```

Comment:
The text should be a label

Reply:
The thing is the checkbox component we are using only accepts label as plain string whilst we need to add line breaks and some other styles to the text.

Reply:
you can wrap the text in a label tag or a fix it in safe-react-components